src/pathtraverser.h

```diff
@@ -4,41 +4,58 @@
 #include <iostream>
 #include <QDir>
 #include <QStringList>
+#include "imageselector.h"
 
 static const QStringList supportedFormats={"jpg","jpeg","png","tif","tiff"};
 
 class MainWindow;
 class PathTraverser
 {
   public:
-    PathTraverser(const std::string path);
+    PathTraverser(const std::string path, bool debugModeIn);
     virtual ~PathTraverser();
     virtual QStringList getImages() const = 0;
     virtual const std::string getImagePath(const std::string image) const = 0;
+    virtual void UpdateOptionsForImage(const std::string& filename, ImageDisplayOptions_t& options) const = 0;
```

Comment:
```suggestion
    virtual const ImageDisplayOptions LoadOptionsForImage(const std::string& filename, const ImageDisplayOptions& options) const = 0;
```

Reply:
Done! Note that makes the pathTraverser->UpdateOptionsForImage() call both pass in options and then update it based on the return value (as I don't want to prevent populateImageDetails() also modifying the options from base settings in the future).